src/data/reducers/assets.js

```diff
@@ -10,13 +11,45 @@ const initialState = {
   direction: 'desc',
 };
 
+const toggleLockAsset = (assets, assetId) => assets.map(asset => (
+  asset.id === assetId ? { ...asset, locked: !asset.locked } : asset));
```

Comment:
Assuming that you need to flip the value because the API value is the opposite of what you want, this is a super simple example of what we could do in a reselect selector. We could keep the state as-is here and compute the value that is necessary for the table in a selector. I don't think this case is worth the hassle of setting it up, but perhaps it gives us some intuition of when we could use it?

Reply:
So, in this case, the asset that is being flipped is the *original* asset. The `PUT`'s response doesn't include the entire asset.

I absolutely agree about evaluating reselect (in the future!).